pkg/datapath/iptables/ipset/ops.go

```diff
@@ -30,37 +31,61 @@ type ops struct {
 	ipset   *ipset
 }
 
-func (ops *ops) Update(ctx context.Context, _ statedb.ReadTxn, entry *tables.IPSetEntry, changed *bool) error {
+// DeleteBatch implements reconciler.BatchOperations.
+func (ops *ops) DeleteBatch(ctx context.Context, txn statedb.ReadTxn, batch []reconciler.BatchEntry[*tables.IPSetEntry]) {
 	if !ops.enabled {
-		return nil
+		return
 	}
 
-	// create the set if does not exist
-	if err := ops.ipset.create(ctx, entry.Name, string(entry.Family)); err != nil {
-		return err
+	addrsByName := map[string][]netip.Addr{}
+	for _, entry := range batch {
+		addrsByName[entry.Object.Name] = append(addrsByName[entry.Object.Name], entry.Object.Addr)
+	}
+	err := ops.ipset.delBatch(ctx, addrsByName)
+	if err != nil {
+		// Fail the whole batch.
+		for i := range batch {
+			batch[i].Result = err
+		}
 	}
+}
 
-	if err := ops.ipset.add(ctx, entry.Name, entry.Addr); err != nil {
-		return err
+// UpdateBatch implements reconciler.BatchOperations.
+func (ops *ops) UpdateBatch(ctx context.Context, txn statedb.ReadTxn, batch []reconciler.BatchEntry[*tables.IPSetEntry]) {
```

Comment:
Differently from before, both `UpdateBatch` and `DeleteBatch` no longer check whether the ipset actually exists before attempting the operations. It looks reasonable to me, especially performance-wise, just making sure that the change was intentional.

Reply:
Yep, that is intentional and it is meant to further improve the performance removing the exec to run `ipset create`. In case of removed sets, we should be able to recover with ta full reconciliation.